System/Console/ANSI/Color.hs

```diff
@@ -0,0 +1,102 @@
+-- | Conversions between color representations
+module System.Console.ANSI.Color
+  (
+  -- | 8-bit to 24-bit color conversion
+    color8To24
+  -- | Utility functions
+  , clamp
+  , color8CodeToEither
+  , closest4bitsANSIColor
+    ) where
+
+
+import qualified Data.Colour.SRGB.Linear as SRGBLin(rgb, Colour)
+import qualified Data.Colour.SRGB as SRGB (RGB (..), toSRGB, Colour)
+
+import Data.Colour.Names (black, blue, cyan, green, grey, lime, magenta, maroon,
+                          navy, olive, purple, red, silver, teal, white, yellow)
+
+import Data.List (minimumBy, (!!))
+
+import System.Console.ANSI.Types
+
+
+color8To24 :: Color8 -> SRGBLin.Colour Float
+color8To24 (RGB8Color r' g' b') =
+  let mapping = (\v -> v/255.0) . fromIntegral . xtermMapRGB8bitComponent . (\v -> clamp v 0 5) :: Int -> Float
```

Comment:
Some explanation would be nice here, too.

Reply:
which aspect exactly would you like to see more documented? the clamp or the division by 255, or both?

Reply:
btw the clamp is also a consequence of `0x10-0xE7: 6 × 6 × 6 cube (216 colors): 16 + 36 × r + 6 × g + b (0 ≤ r, g, b ≤ 5)`, maybe I can mention that in the Module doc to avoid repeating the same doc at several places...